pkg/model/dns.go

```diff
@@ -91,6 +91,29 @@ func (b *DNSModelBuilder) Build(c *fi.ModelBuilderContext) error {
 		}
 	}
 
+	//TODO: use interface (but what about bastion?), should I create a new member instead of targetLoadBalancer for the interface?
```

Comment:
I definitely like the idea of an interface implemented by both LoadBalancer and NetworkLoadBalancer. It will need to return at least the DNSName and HostedZoneId and would allow simplifying a bunch of logic.

Reply:
There are other situations such as in spot, where an interface for either Loadbalancer or NetworkLoadBalancer would help. I wonder if DNSTarget could be re-used in all instances, or if it would be more semantically correct to have a LB interface as well separate from DNSTarget.  Methods needed in the spot instance would be getLoadBalancerName getLoadBalancerArn, getLoadBalancerType